moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -182,25 +190,79 @@ void ServoCalcs::start()
   initial_joint_trajectory->points.push_back(point);
   last_sent_command_ = initial_joint_trajectory;
 
-  timer_ = nh_.createTimer(period_, &ServoCalcs::run, this);
-
   current_state_ = planning_scene_monitor_->getStateMonitor()->getCurrentState();
   tf_moveit_to_ee_frame_ = current_state_->getGlobalLinkTransform(parameters_.planning_frame).inverse() *
                            current_state_->getGlobalLinkTransform(parameters_.ee_frame_name);
   tf_moveit_to_robot_cmd_frame_ = current_state_->getGlobalLinkTransform(parameters_.planning_frame).inverse() *
                                   current_state_->getGlobalLinkTransform(parameters_.robot_link_command_frame);
+
+  stop_requested_ = false;
+  thread_ = std::thread([this] { main(); });
 }
 
-void ServoCalcs::run(const ros::TimerEvent& timer_event)
+void ServoCalcs::stop()
```

Comment:
This overlaps with the existing functionality, `setPaused()`

Reply:
It is a private function and not part of the exposed interface.  I created this function because I needed the block of code in the destructor and in start().